File: app/src/main/java/com/stevenfrew/beatprompter/filter/FolderFilter.java
package com.stevenfrew.beatprompter.filter;

import com.stevenfrew.beatprompter.cache.SongFile;

import java.util.ArrayList;

public class FolderFilter extends SongFilter
{
    public FolderFilter(String folderName,ArrayList<SongFile> songs)
    {
        super(folderName,songs,true);
    }

    public boolean equals(Object o) {
        return o instanceof FolderFilter && mName.equals(((FolderFilter) o).mName);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/filter/FolderFilter.kt
package com.stevenfrew.beatprompter.filter

import com.stevenfrew.beatprompter.cache.SongFile
import java.util.ArrayList

class FolderFilter(folderName: String, songs: ArrayList<SongFile>) : SongFilter(folderName, songs, true) {

    override fun equals(other: Any?): Boolean {
        return other is FolderFilter && mName == other.mName
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}